src/storage/src/statistics.rs

```diff
@@ -83,6 +87,16 @@ impl SourceStatisticsMetricDefs {
                 help: "The amount of time in milliseconds it took for the worker to rehydrate the source envelope state. This will be specific to the envelope in use.",
                 var_labels: ["source_id", "worker_id", "parent_source_id", "shard_id", "envelope"],
             )),
+            upstream_values: registry.register(metric!(
+                name: "mz_source_upstream_values",
+                help: "The total number of _values_ (source-defined unit) present in upstream.",
+                var_labels: ["source_id", "worker_id", "shard_id"],
+            )),
+            processed_values: registry.register(metric!(
+                name: "mz_source_processed_values",
+                help: "The total number of _values_ (source-defined unit) we have fully processed.",
```

Comment:
Since this is driven by the resume uppers and "processed" is a bit vague should rename it to `committed` and in the help text explicitly mention that this counts the number of values committed to persist?

Reply:
sounds good!